cmake/cmakepp_core/utilities/pack_list.cmake

```diff
@@ -0,0 +1,146 @@
+include_guard()
+include(cmakepp_core/serialization/serialization)
+
+#[[[ Transforms a CMake list into a string that can be passed down through
+#    function calls and then transformed back to the original CMake list that
+#    retains the original nesting structure of the list.
+#
+# This function will take a CMake list and replace list separator characters at
+# various levels of nesting with "_CPP_{N}_CPP_" where N is the level of
+# nesting. N will be 0 for the top level list, 1 for the next level down, and so
+# on.
+#
+# For example, take a list declared with the command:
+# set(my_list a;b\\\;c\\\\\;cc\\\;bb;aa)
+#
+# my_list serializes to the (much more easily read) form:
+# [ "a", [ "b", [ "c", "cc" ], "bb" ], "aa" ]
+#
+# my_list will be transformed to the packed list string:
+# a_CPP_0_CPP_b_CPP_1_CPP_c_CPP_2_CPP_cc_CPP_1_CPP_bb_CPP_0_CPP_aa
+#
+# :param _p_result: The name for the variable which will hold the result.
+# :type _p_result: desc
+# :param _p_list: The list we want to pack into a string.
+# :type _p_result: list
+# :returns: ``_p_result`` will be set to the resulting packed list string.
+# :rtype: desc
+#]]
+function(cpp_pack_list _p_result _p_list)
+    # Get the depth argument, if no depth was set, set it to 0
+    cmake_parse_arguments(_p "" DEPTH "" ${ARGN})
+    if(NOT _p_DEPTH)
+        set(_p_DEPTH 0)
+    endif()
+
+    # Start the result string
+    set(_p_packed_list "")
+
+    # Start a counter and capture length of list
+    set(_p_counter 0)
+    list(LENGTH _p_list _p_n_elems)
+
+    # Calc next depth
+    math(EXPR _p_next_depth "${_p_DEPTH} + 1")
+
+    # Loop over elements in list
+    foreach(_p_elem ${_p_list})
+        # Get the type of current element
+        cpp_type_of(_p_elem_type "${_p_elem}")
+        if(_p_elem_type STREQUAL "list")
+            # If element is a list, recursively pack it and add it to result
+            cpp_pack_list(_p_elem_result "${_p_elem}" DEPTH ${_p_next_depth})
+            string(APPEND _p_packed_list "${_p_elem_result}")
+        else()
+            # If the element is not a list, just add it to result
+            string(APPEND _p_packed_list "${_p_elem}")
+        endif()
+
+        # Update counter
+        math(EXPR _p_counter "${_p_counter} + 1")
+        if("${_p_counter}" LESS "${_p_n_elems}")
+            # Add a seperation character if this is not the last item
+            string(APPEND _p_packed_list "_CPP_${_p_DEPTH}_CPP_")
```

Comment:
I'd recommend trying the `string(ASCII 65 output)` command and seeing if `output` is set to capital `A`. If it is, then I'd recommend doing:
```
string(ASCII 04 delim)
string(APPEND _pl_packed_list "${delim}_${pl_DEPTH}_${delim}")
```
which will use the ASCII end-of-transmission character as the separator instead of `_CPP`. I use `_CPP` internally as a prefix on some variables so it's conceivable it may conflict, but I highly doubt anyone is going to use the end-of-transmission character (unless they're being malicious)